### PR TITLE
[sqllab] surfacing more table metadata (indices, pk, fks)

### PR DESCRIPTION
**Pull request by mistercrunch**
screen shot 2016-10-30 at 4 59 56 pm

@vera-liu @ascott @bkyryliuk

### CODE REVIEW

File: caravel/views.py [hunk 1]
```diff
@@ -1947,13 +1947,24 @@ def table(self, database_id, table_name, schema):
...
         except Exception as e:
             return Response(
                 json.dumps({'error': utils.error_msg_from_exception(e)}),
                 mimetype="application/json")
-        indexed_columns = set()
-        for index in indexes:
-            indexed_columns |= set(index.get('column_names', []))
+        keys = []
+        if primary_key and primary_key.get('constrained_columns'):
+            primary_key['column_names'] = primary_key.pop('constrained_columns')
```
Comment (bkyryliuk) [inline caravel/views.py]: pep8 nit, line len should be 79

File: tests/core_tests.py [hunk 2]
```diff
@@ -519,6 +519,15 @@ def test_templated_sql_json(self):
         data = self.run_sql(sql, "admin", "fdaklj3ws")
         self.assertEqual(data['data'][0]['test'], "2017-01-01T00:00:00")
 
+    def test_table_metadata(self):
+        dbid = self.get_main_database(db.session).id
+        data = self.get_json_resp(
+            "/caravel/table/{}/ab_user/null/".format(dbid))
+        self.assertEqual(data['name'], 'ab_user')
+        assert len(data['columns']) > 5
```
Comment (bkyryliuk) [inline tests/core_tests.py]: could you please extend this test and check in case of MySQL db engine that the code returns: index, pk and fk ?

**Comment by bkyryliuk** [comment]
not sure about black background for the modal window

**Comment by bkyryliuk** [comment]
LGTM, once the test covers the retrieval of the `pk`, `fk` and `index` for mysql engine.

File: caravel/views.py (hunk 1 above)
Comment (mistercrunch) [inline caravel/views.py]: I need to find how to enable up to 90 on codeclimate...

File: tests/core_tests.py (hunk 2 above)
Comment (mistercrunch) [inline tests/core_tests.py]: I thought about it and didn't want to write backend specific tests but I guess it's not very hard. I'll add mysql/postgres-specific assertions

**Comment by mistercrunch** [comment]
The black color is the normal color for tooltip on our theme. I thought of using popovers but tooltips work pretty well

**Pull request merged**